src/persist-client/src/internal/trace.rs

```diff
@@ -103,10 +112,225 @@ impl<T: Timestamp + Lattice> Default for Trace<T> {
     fn default() -> Self {
         Self {
             spine: Spine::new(),
+            roundtrip_structure: false,
         }
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ThinSpineBatch<T> {
+    pub(crate) level: usize,
+    pub(crate) desc: Description<T>,
+    pub(crate) parts: Vec<SpineId>,
+}
+
+#[derive(Clone, Debug)]
+pub struct FlatTrace<T> {
```

Comment:
We discussed offline that it would be good to type out the various considerations that led us to this particular structure. I think FlatTrace is my vote for where to hang that comment off of

Reply:
I think I've captured everything we've discussed - feel free to unresolve this if that's not right!